librz/analysis/arch/ppc/ppc_il_ops.c

```diff
@@ -0,0 +1,1587 @@
+	case PPC_INS_LHA:
+	case PPC_INS_LHAX:
+	case PPC_INS_LHAU:
+	case PPC_INS_LHAUX:
+	case PPC_INS_LWA:
+	case PPC_INS_LWAX:
+	case PPC_INS_LWAUX:
+	case PPC_INS_LBZCIX:
+	case PPC_INS_LHZCIX:
+	case PPC_INS_LWZCIX:
+	case PPC_INS_LDCIX:
+		base = VARG(rA);
+		if (ppc_is_x_form(id)) {
+			disp = VARG(rB);
+		} else {
+			// "disp << 2" done by capstone.
+			RzILOpPure *imm = SN(16, d);
+			disp = EXTEND(PPC_ARCH_BITS, imm);
+		}
+		ea = ADD(base, disp);
+		RzILOpPure *loadw = LOADW(mem_acc_size, VARLP("ea"));
+		if (ppc_is_algebraic(id)) {
+			into_rt = EXTEND(PPC_ARCH_BITS, VARLP("loadw"));
+		} else {
+			into_rt = EXTZ(VARLP("loadw"));
+		}
+		into_rt = LET("ea", ea, LET("loadw", loadw, into_rt));
+		break;
+	case PPC_INS_LWARX:
+	case PPC_INS_LDARX:
+		NOT_IMPLEMENTED;
+	// Byte reverse and reserved indexed
+	case PPC_INS_LHBRX:
+	case PPC_INS_LWBRX:
+	case PPC_INS_LDBRX:
+		base = VARG(rA);
+		disp = VARG(rB);
+		ea = ADD(base, disp);
+
+#define LB(i) LOADW(8, ADD(VARLP("ea"), UA(i)))
+		if (IN_BE_MODE) {
+			if (id == PPC_INS_LHBRX) {
+				into_rt = LET("ea", ea, APPEND(LB(0), LB(1)));
+			} else if (id == PPC_INS_LWBRX) {
+				into_rt = LET("ea", ea, APPEND(LB(0), APPEND(LB(1), APPEND(LB(2), LB(3)))));
+			} else {
+				into_rt = LET("ea", ea, APPEND(LB(0), APPEND(LB(1), APPEND(LB(2), APPEND(LB(3), APPEND(LB(4), APPEND(LB(5), APPEND(LB(6), LB(7)))))))));
+			}
+		} else {
+			if (id == PPC_INS_LHBRX) {
+				into_rt = LET("ea", ea, APPEND(LB(1), LB(0)));
+			} else if (id == PPC_INS_LWBRX) {
+				into_rt = LET("ea", ea, APPEND(LB(3), APPEND(LB(2), APPEND(LB(1), LB(0)))));
+			} else {
+				into_rt = LET("ea", ea, APPEND(LB(7), APPEND(LB(6), APPEND(LB(5), APPEND(LB(4), APPEND(LB(3), APPEND(LB(2), APPEND(LB(1), LB(0)))))))));
+			}
+		}
+		into_rt = EXTZ(into_rt);
+		break;
+	// Floats
+	case PPC_INS_LFD:
+	case PPC_INS_LFDX:
+	case PPC_INS_LFDU:
+	case PPC_INS_LFDUX:
+	case PPC_INS_LFIWAX:
+	case PPC_INS_LFIWZX:
+	case PPC_INS_LFS:
+	case PPC_INS_LFSX:
+	case PPC_INS_LFSU:
+	case PPC_INS_LFSUX:
+		NOT_IMPLEMENTED;
+	// Vector
+	case PPC_INS_LVEBX:
+	case PPC_INS_LVEHX:
+	case PPC_INS_LVEWX:
+	case PPC_INS_LVSL:
+	case PPC_INS_LVSR:
+	case PPC_INS_LVX:
+	case PPC_INS_LVXL:
+		NOT_IMPLEMENTED;
+	// String word
+	case PPC_INS_LSWI:
+		NOT_IMPLEMENTED;
+	// VSX Scalar
+	case PPC_INS_LXSDX:
+	// VSX Vector
+	case PPC_INS_LXVD2X:
+	case PPC_INS_LXVDSX:
+	case PPC_INS_LXVW4X:
+		NOT_IMPLEMENTED;
+	}
+
+	rz_return_val_if_fail(into_rt, NULL);
+	RzILOpEffect *res = SETG(rT, into_rt);
+	RzILOpEffect *update = update_ra ? SETG(rA, DUP(ea)) : EMPTY();
+	return SEQ2(res, update);
+}
+
+/**
+ * \brief Determine log_2( \p v ). \p v Should be a multiple of 2.
+ * See: https://graphics.stanford.edu/~seander/bithacks.html#IntegerLogObvious
+ *
+ * \param v Value to determine log_2(v) for.
+ * \return ut32 lg(v)
+ */
+static inline ut32 ppc_log_2(const ut32 v) {
+	ut32 x = v;
+	ut32 r = 0; // r will be lg(v)
+	while (x >>= 1) {
+		r++;
+	}
+	return r;
+}
```

Comment:
Please add a comment above to simplify this using some fast log2 approximation algorithm.